### DISCUSSION TITLE
Can LosslessCut replicate what VideoRedo can do w/frame accurate cuts and joined/merged files?

### DISCUSSION THREAD

**Discussion by sou7611** [mifi/lossless-cut#2148] [Support]
VideoRedo is a now-defunct piece of software that is essentially the same thing that LosslessCut is moving towards being ... a frame accurate editing tool for trimming video files ... typically for lopping un-needed content off the front/back of a file, or cutting out commercials.

But the frame accuracy on cuts is ... iffy.  Very iffy.  Any ideas on when the Smart Cut functionality will be further perfected?

And it appears to be missing a crucial key feature, that being the ability to dump segments into a joiner list to merge into a single file.

In VRD there's a "joiner" function similar to the LosslessCut Batch where you can add segments to be joined/merged, and they output to an edited file.  It's slick as hell and it works flawlessly, producing an output file that works on any video player.

Near as I can tell, Batch only merges actual files; it ignores any segments/edits that you've marked within the files listed.

If this kind of functionality can be ironed out, there's a serious market to be exploited : *nothing* else (other than VRD) does this kind of editing

**Comment by mifi** [comment]
thanks for the info, I've updated what i've found here in the OP: #126

**Comment by glenpinn** [comment]
VRD is not defunct yet, so please stop voicing this rubbish, VRD was, and still is a 1 of a kind Frame Accurate, lossless cutting tool AND Encoder that utilizes the built in Smart Render and the unique Intelligent Recode features, where you can take your imported or edited files and output to just about any other format and resolution.

Unfortunately because of the owners death 2 years ago, the software will never be updated, we have to live with the build that we all had at the time of Dan's passing, but VRD can still be used until further notice, it all depends on what happens to the Licensing/Activation servers.

Losslesscut seems to only cut and join accurately on Keyframes from my own testing, and i can live with this as my cuts don't need to be that precise, but Losslesscut cannot recode cut files to other Formats anywhere near the same way as VRD does it, it works completely different, and that is why Losslesscut is not, nor will it likely be a replica of VRD as a unique, all in one frame accurate cutting and recoding tool.

Unfortunately for me, now that i record my videos in 4k/60p HEVC (60 and 144Mbps) VRD stutters and gets laggy when i try to play these files in the preview player, and also does the same when seeking my cut points in the timeline, and it was almost impossible for me to do my cuts using VRD, so i had to try and find an alternative frame accurate cutter that did not have this jitter or lag in the preview player and when seeking my cut points.

Thankfully i found Losslesscut, and this cutting tool does not get jittery or laggy like it does with VRD Pro, it is very smooth, and also very fast as well, to the point where i now use Losslesscut to cut my files (only on keyframes) and smart render them back to the original format before taking the edited files into my VRD Pro for processing to other formats if needed, or to add Tiles etc.

If the Developer of Losslesscut can achieve exactly what Dan Rosen was able to do with VRD, then i will take my hat off to them, for now, i take my hat off for being able to use Losslesscut for my actual edits and have it done fast and smoothly.

Cheers

**Comment by AmBintroGoof99** [comment]
Came here to add that, at least from my own experience, LosslessCut cuts accurately not on “keyframe cut” mode, and not on “Smart Cut” mode, but on the “Normal Cut” mode. So much so that the output is almost identical to the ones produced by VideoReDo TVSuite. I made the cuts on the same video, with the exact same cut points, which were not on keyframes. It’s not perfect, obviously, but very impressive. One big problem, though, is that LosslessCut’s outputs have extremely high audio “delay relative to video” values, which VideoReDo and Avidemux usually have minimal of…and even if they do, a quick remux brings that value down to nothing.

**Comment by SCAR2195** [comment]
I'm a long-time VRD user and have been using LosslessCut for videos that are either 2k/4k or have subtitles, but I have yet to get Smart cut to work properly when using frame-accurate set points instead of keyframes - I keep getting black video at the beginning and end of the output, and the subtitles are not sync'd properly.

AmBintroGoog99 - are you getting proper frame-accurate cuts by setting both 'Smart cut' and 'Keyframe cut mode' to off?

**Reply by AmBintroGoof99** [reply]
> I'm a long-time VRD user and have been using LosslessCut for videos that are either 2k/4k or have subtitles, but I have yet to get Smart cut to work properly when using frame-accurate set points instead of keyframes - I keep getting black video at the beginning and end of the output, and the subtitles are not sync'd properly.
> 
> 
> 
> AmBintroGoog99 - are you getting proper frame-accurate cuts by setting both 'Smart cut' and 'Keyframe cut mode' to off? 

Yes, on par with VDRD. However, some files either take way too long or are simply not cut with "Normal Cut" on. This is quite rare, though. At this point, the only thing I've found that sets LosslessCut apart from VDRD is that VDRD handles also the audio delay issues, while LC doesn't.

**Comment by SCAR2195** [comment]
Good to know, thanks!

Will give that a shot : )

**Comment by andcbii** [comment]
I'm not able to cut shows correctly. 

"Smart cut (experimental)" = ON
"Smart cut auto detect bitrate" = ON

The audio is off where the it joins the two segments

"Smart cut (experimental):" = OFF
"Keyframe cut mode" = OFF

It almost looks like the file is cut correctly, but payback starts at the next keyframe and not at the place i put the cut mark. When the file starts to play it will show as already 1-3 seconds into the video and you can not actually go back to the 0.0 time code.

**Reply by AmBintroGoof99** [reply]
> I'm not able to cut shows correctly. 
> 
> 
> 
> "Smart cut (experimental)" = ON
> 
> "Smart cut auto detect bitrate" = ON
> 
> 
> 
> The audio is off where the it joins the two segments
> 
> 
> 
> "Smart cut (experimental):" = OFF
> 
> "Keyframe cut mode" = OFF
> 
> 
> 
> It almost looks like the file is cut correctly, but payback starts at the next keyframe and not at the place i put the cut mark. When the file starts to play it will show as already 1-3 seconds into the video and you can not actually go back to the 0.0 time code. 

Maybe remuxing the file will help. Anyways, what matters is that Normal Cut's outputs are the closest you can get to frame-accurate. I'd report your experience as a bug.

**Reply by AmBintroGoof99** [reply]
My experience is different from yours. Yes, there are times when the start point is a frame ahead of the one you went for, but this has only happened with one file.
